File: backend/SNS.Domain/Search/Documents/CommunityDocument.cs
using SNS.Domain.ContentManagement.Communities.Enums;

namespace SNS.Domain.Search.Documents;

public class CommunityDocument
{
    public Guid Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public CommunityType Type { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}

File: WebClient/src/app/communities/dtos/community.document.ts
import { CommunityType } from "../enums/community.type.enum"

export interface CommunityDocument {
    id: string
    name: string
    description: string
    logoUrl: string
    type: CommunityType
    membersCount: number
    createdAt: Date;
    ownerId: string;
    ownerName: string;
}